src/VestingWalletWithCliffAndClawback.sol

```diff
@@ -53,166 +27,42 @@ contract VestingWalletWithCliffAndClawback is VestingWallet, Ownable2Step {
         uint64 durationSeconds,
         uint64 cliffDurationSeconds
     ) payable 
-        Ownable2Step()
         VestingWallet(beneficiaryAddress, startTimestamp, durationSeconds)
+        VestingWalletWithCliff(cliffDurationSeconds)
+        VestingWalletWithClawback(ownerAddress)
     {
-        _cliffDuration = cliffDurationSeconds;
-        _transferOwnership(ownerAddress);
-    }
-
-    /**
-     * @dev Getter for the vesting cliff duration.
-     */
-    function cliffDuration() public view virtual returns (uint256) {
-        return _cliffDuration;
-    }
-
-    /**
-     * @dev Getter for whether a native asset clawback has occurred.
-     */
-    function clawbackHasOccurred() public view virtual returns (bool) {
-        return _clawbackHasOccurred;
-    }
-
-    /**
-     * @dev Getter for whether a token clawback has occurred.
-     */
-    function clawbackHasOccurred(address token) public view virtual returns (bool) {
-        return _erc20ClawbackHasOccurred[token];
-    }
-
-    /**
-     * @dev Allow owner to clawback unvested native assets from contract.
-     *
-     * Emits an {EtherClawedBack} event.
-     */
-    function clawback() public onlyOwner {
-        if (clawbackHasOccurred()) {
-            revert ClawbackHasAlreadyOccurred();
-        }
-
-        uint256 releasableNativeAsset = releasable();
-
-        // Store the max cumulative payout to recipient after the the clawback has occurred
-        // Need to store value as cumulative value because `release` only modifies `_released`
-        _cumulativeReleasablePostClawback = released() + releasableNativeAsset;
-
-        // Log that the clawback has occurred
-        _clawbackHasOccurred = true;
-
-        // Send current balance less current redeemable amount back to owner
-        uint256 amount = address(this).balance - releasableNativeAsset;
-        Address.sendValue(payable(msg.sender), amount);
-        emit EtherClawedBack(amount);
     }
 
     /**
-     * @dev Allow owner to clawback unvested ERC20 tokens from contract.
-     *
-     * Emits an {ERC20ClawedBack} event.
+     * @dev Override for `releasable()` to stack logic respectively from VestingWalletWith{Cliff,Clawback}.
```

Comment:
Because of the way the diff is formatted, I can't make a direct suggestion here, but I would simply replace this NatSpec with:

```solidity
/// @inheritdoc VestingWalletWithClawback
```

This makes it clear to the reader which implementation you are effectively using, and tells them where to go to read the relevant comment.

Reply:
done